File: hello-streams-spring/src/main/java/me/zixuan/kafkastream/hellostramsspring/processor/KafkaProcessor.java
package me.zixuan.kafkastream.hellostramsspring.processor;

import lombok.val;
import org.apache.kafka.streams.StreamsBuilder;
import org.apache.kafka.streams.kstream.KStream;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

@Component
public class KafkaProcessor {

    @Bean
    public KStream<Void, String> kStream(StreamsBuilder kStreamBuilder) {
        val TOPIC_NAME = "test";
        KStream<Void, String> stream = kStreamBuilder.stream(TOPIC_NAME);
        stream.foreach((key, value) -> System.out.println("Received message: " + value));
        return stream;
    }

}

File: hello-streams-spring-kt/src/main/kotlin/me/zixuan/kafkastream/hellostreamsspringkt/processor/KafkaProcessor.kt
package me.zixuan.kafkastream.hellostreamsspringkt.processor

import org.apache.kafka.streams.StreamsBuilder
import org.apache.kafka.streams.kstream.KStream
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component

@Component
class KafkaProcessor {
    @Bean
    fun kStream(kStreamsBuilder: StreamsBuilder): KStream<Void, String> {
        val topicName = "test"
        val stream = kStreamsBuilder.stream<Void, String>(topicName)
        stream.foreach { _, value -> println("Received message: $value") }
        return stream
    }
}